pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -338,15 +340,110 @@ func (pl *VolumeRestrictions) EventsToRegister() []framework.ClusterEventWithHin
 	}
 }
 
+// isSchedulableAfterPodDeleted is invoked whenever a pod deleted,
+// It checks whether the deleted pod will conflict with volumes of other pods on the same node
+func (pl *VolumeRestrictions) isSchedulableAfterPodDeleted(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	oldPod, _, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterPodDeleted", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	readWriteOncePVCUseByAnotherPod, err := pl.readWriteOncePodPVCUseByAnotherPod(context.TODO(), pod)
+	if err != nil {
+		logger.Error(err, "encountered an error when obtaining PVC according to the pod", "pod", klog.KObj(pod))
+	}
+
+	if pl.enableReadWriteOncePod && readWriteOncePVCUseByAnotherPod {
+		logger.V(4).Info("The Pod uses a PVC with ReadWriteOncePod access mode that is occupied by another Pod.", "pod", klog.KObj(pod))
+		return framework.QueueAfterBackoff
+	}
+
+	nodeInfo := framework.NewNodeInfo(oldPod)
+	if !satisfyVolumeConflicts(pod, nodeInfo) {
+		return framework.QueueAfterBackoff
+	}
+
+	return framework.QueueSkip
+}
+
+// isSchedulableAfterNodeAdded is invoked whenever a node added, It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeRestrictions) isSchedulableAfterNodeAdded(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	_, newNode, err := util.As[*v1.Node](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterNodeAdded", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	readWriteOncePVCUseByAnotherPod, err := pl.readWriteOncePodPVCUseByAnotherPod(context.TODO(), pod)
+	if err != nil {
+		logger.Error(err, "encountered an error when obtaining PVC according to the pod", "pod", klog.KObj(pod))
+	}
+
+	if pl.enableReadWriteOncePod && readWriteOncePVCUseByAnotherPod {
+		logger.V(4).Info("The Pod uses a PVC with ReadWriteOncePod access mode that is occupied by another Pod.", "pod", klog.KObj(pod))
+		return framework.QueueAfterBackoff
+	}
+
+	logger.V(4).Info("node was added", "pod", klog.KObj(pod), "node", klog.KObj(newNode))
+	return framework.QueueSkip
+}
+
+// isSchedulableAfterPersistentVolumeClaimChange is invoked whenever a PersistentVolumeClaim changed, It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeRestrictions) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	_, _, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterPersistentVolumeClaimChange", "oldObj", oldObj, "newObj", newObj)
```

Comment:
I don't think this check is needed, same as `isSchedulableAfterNodeAdded`, but I'll leave a discussion here.

Reply:
Yes, I still think we need to do checks here to avoid unanticipated errors

Reply:
Generally speaking, if we don't need the typed object, we don't need to try to convert.
Unnecessary conversion would just take additional time.

But, as writing in another comment, we need a typed one anyways to check further stuff.